transport/internal/errors/peers.go

```diff
@@ -94,3 +94,43 @@ type ErrInvalidAgentConversion struct {
 func (e ErrInvalidAgentConversion) Error() string {
 	return fmt.Sprintf("cannot convert agent (%v) to type %s", e.Agent, e.ExpectedType)
 }
+
+// ErrNoPeerToSelect is used when a peerlist doesn't have any peers to return
+type ErrNoPeerToSelect string
+
+func (e ErrNoPeerToSelect) Error() string {
+	return fmt.Sprintf("could not find a peer to select in peerlist %s", string(e))
+}
+
+// ErrPeerAlreadyInList is used when a peerlist attempts to add a peer that is already in the list
```

Comment:
returned to peer providers if the peerlist is already tracking a peer for the added identifier